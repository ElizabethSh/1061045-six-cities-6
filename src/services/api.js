import axios from 'axios';

const TIMEOUT = 5000;
const BASE_URL = `https://6.react.pages.academy/six-cities`;

const statusCode = {
  UNAUTORIZED: 401,
};

export const createAPI = (unAuthorized) => {
  const api = axios.create({
    baseURL: BASE_URL,
    tiomeout: TIMEOUT,
    withCredentials: true,
  });

  const onSuccess = (response) => response;

  const onError = (error) => {
    const {response} = error;

    if (response.status === statusCode.UNAUTORIZED) {
      unAuthorized();

      throw error;
    }

    throw error;
  };

  axios.interceptors.response.use(onSuccess, onError);

  return api;
};

