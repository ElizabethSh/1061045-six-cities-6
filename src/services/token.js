const LOCALSTORAGE_AUTH_TOKEN = "6-sities-auth-token";

const isLocalStorageAvailable = () =>
  typeof window !== "undefined" && !!window.localStorage;

export const getToken = () => {
  try {
    if (
      isLocalStorageAvailable() &&
      typeof window.localStorage.getItem === "function"
    ) {
      const token = localStorage.getItem(LOCALSTORAGE_AUTH_TOKEN);
      return token ? JSON.parse(token) : undefined;
    }
    return undefined;
  } catch (e) {
    return undefined;
  }
};

export const setToken = (token) => {
  try {
    if (
      isLocalStorageAvailable() &&
      typeof window.localStorage.setItem === "function"
    ) {
      localStorage.setItem(LOCALSTORAGE_AUTH_TOKEN, JSON.stringify(token));
    }
    // eslint-disable-next-line no-empty
  } catch (e) {}
};

export const dropToken = () => {
  try {
    if (
      isLocalStorageAvailable() &&
      typeof window.localStorage.removeItem === "function"
    ) {
      localStorage.removeItem(LOCALSTORAGE_AUTH_TOKEN);
    }
    // eslint-disable-next-line no-empty
  } catch (e) {}
};
