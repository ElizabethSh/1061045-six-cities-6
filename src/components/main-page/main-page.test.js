import React from 'react';
import {render} from '@testing-library/react';
import configureStore from 'redux-mock-store';
import {createMemoryHistory} from "history";
import {Provider} from 'react-redux';
import MainPage from './main-page';
import {Router} from 'react-router';
import thunk from 'redux-thunk';
import {createAPI} from '../../services/api';

const api = createAPI(() => {});
const mockStore = configureStore([thunk.withExtraArgument(api)]);
let history;

describe(`Test 'MainPage'`, () => {
  beforeEach(() => {
    history = createMemoryHistory();
  });
  it(`MainPage should render 'Loader' when data is loading`, () => {
    const store = mockStore({
      OFFER: {isOffersLoaded: false, offers: []}
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <MainPage/>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it(`MainPage should render empty page if there are no offers`, () => {
    const store = mockStore({
      OFFER: {isOffersLoaded: true, offers: []},
      USER: {isLoggedIn: false},
    });

    const {container} = render(
        <Provider store={store}>
          <Router history={history}>
            <MainPage/>
          </Router>
        </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
