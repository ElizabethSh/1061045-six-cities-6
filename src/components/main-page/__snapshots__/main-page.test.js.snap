// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'MainPage' MainPage should render 'Loader' when data is loading 1`] = `
<div>
  <div
    class="loader"
  >
    <div
      class="loader__dual-ring"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;

exports[`Test 'MainPage' MainPage should render empty page if there are no offers 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
                style="display: flex;"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  />
                  <span
                    class="header__login"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index
        page__main--index-empty"
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item
                  "
                href="/city/Paris"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item
                  "
                href="/city/Cologne"
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item
                  "
                href="/city/Brussels"
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item
                  "
                href="/city/Amsterdam"
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item
                  "
                href="/city/Hamburg"
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item
                  "
                href="/city/Dusseldorf"
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container cities__places-container--empty container"
        >
          <section
            class="cities__no-places"
          >
            <div
              class="cities__status-wrapper tabs__content"
            >
              <b
                class="cities__status"
              >
                No places to stay available
              </b>
              <p
                class="cities__status-description"
              >
                We could not find any property available at the moment in undefined
              </p>
            </div>
          </section>
          <div
            class="cities__right-section"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
